odoo/addons/base/models/ir_ui_view.py

```diff
@@ -1564,14 +1564,20 @@ def _validate_tag_field(self, node, name_manager, node_info):
         name_manager.has_field(node, name, {'id': node.get('id'), 'select': node.get('select')})
 
         if validate:
+            class dict_get_bool(dict):
+                def __getitem__(self, *args, **kwargs):
+                    return bool(super().__getitem__(*args, **kwargs))
+                def get(self, *args, **kwargs):
+                    return bool(super().get(*args, **kwargs))
             for attribute in ('invisible', 'readonly', 'required'):
                 val = node.get(attribute)
                 if val:
                     try:
                         # most (~95%) elements are 1/True/0/False
                         res = str2bool(val)
                     except ValueError:
-                        res = safe_eval.safe_eval(val, {'context': self._context})
+                        context = dict_get_bool(**self._context)
+                        res = safe_eval.safe_eval(val, {'context': context})
```

Comment:
What happens if one wants to do `invisible="context.get('lang') == 'en_US'"` or `invisible="context.get('active_model', 'crm.lead') != 'crm.lead'"`?